### PR TITLE
Mbst 17375/migrate to application service client

### PR DESCRIPTION
**Pull request by spraxxy**
Removed references to current system of Application and ApplicationConfiguration and replaced them with ApplicationsServiceClient

### CODE REVIEW

File: src/main/java/org/atlasapi/equiv/update/updaters/providers/item/VodItemWithSeriesSequenceUpdaterProvider.java [hunk 1]
```diff
@@ -119,4 +123,8 @@ public static VodItemWithSeriesSequenceUpdaterProvider create() {
                 )
                 .build();
     }
+
+    private Application defaultAppWithSources(Iterable<Publisher> publishers) {
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/equiv/update/updaters/providers/item/VodItemWithSeriesSequenceUpdaterProvider.java]: This method is duplicated across multiple classes, can they be extracted somewhere?

File: src/main/java/org/atlasapi/output/DispatchingAtlasModelWriter.java [hunk 2]
```diff
@@ -62,10 +56,10 @@ public void writeError(HttpServletRequest request, HttpServletResponse response,
 	}
 
     @Override
-    public void writeTo(HttpServletRequest request, HttpServletResponse response, T graph, Set<Annotation> annotations, ApplicationConfiguration config) throws IOException {
+    public void writeTo(HttpServletRequest request, HttpServletResponse response, T graph, Set<Annotation> annotations, Application application) throws IOException {
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/output/DispatchingAtlasModelWriter.java]: line length

File: src/main/java/org/atlasapi/output/DispatchingAtlasModelWriter.java [hunk 3]
```diff
@@ -109,12 +103,12 @@ public void writeNotFound(HttpServletResponse response) throws IOException {
 		}
 		
         @Override
-        public void writeTo(HttpServletRequest request, HttpServletResponse response, T graph, Set<Annotation> annotations, final ApplicationConfiguration config) throws IOException {
+        public void writeTo(HttpServletRequest request, HttpServletResponse response, T graph, Set<Annotation> annotations, final Application application) throws IOException {
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/output/DispatchingAtlasModelWriter.java]: line length

File: src/main/java/org/atlasapi/output/JsonTranslator.java [hunk 4]
```diff
@@ -78,7 +78,13 @@ public JsonTranslator(GsonBuilder gsonBuilder) {
     }
 
     @Override
-	public void writeTo(HttpServletRequest request, HttpServletResponse response, T model, Set<Annotation> annotations, ApplicationConfiguration config) throws IOException {
+	public void writeTo(
+			HttpServletRequest request,
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/output/JsonTranslator.java]: weird indentation

File: src/main/java/org/atlasapi/output/SimplePersonModelWriter.java [hunk 5]
```diff
@@ -23,22 +22,25 @@
...
+            AvailableItemsResolver availableResolver
+    ) {
 		super(outputter);
         this.personSimplifier = new PersonModelSimplifier(imageSimplifier, upcomingResolver, availableResolver);
 	}
 	
 	@Override
-	protected PeopleQueryResult transform(Iterable<Person> people, final Set<Annotation> annotations, final ApplicationConfiguration config) {
+	protected PeopleQueryResult transform(
+	        Iterable<Person> people,
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/output/SimplePersonModelWriter.java]: weird indentation

File: src/main/java/org/atlasapi/query/v2/EventsController.java [hunk 6]
```diff
@@ -119,13 +112,19 @@ public void allEvents(HttpServletRequest request, HttpServletResponse response,
...
-            Predicate<Event> eventFilter = isEnabled(appConfig);
+            Predicate<Event> eventFilter = isEnabled(application);
             if (!Boolean.TRUE.equals(noWhitelist)) {
                 eventFilter = Predicates.and(eventFilter, filterNonWhitelistedEvents); 
             }
-            events = selection.apply(Iterables.filter(eventResolver.fetch(eventGroup, from), eventFilter));
+            events = selection.apply(
+                    StreamSupport.stream(
+                            eventResolver.fetch(eventGroup, from).spliterator(),
+                            false)
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/EventsController.java]: Indentation is a bit weird here

File: src/main/java/org/atlasapi/query/v2/PeopleWriteController.java [hunk 7]
```diff
@@ -77,25 +76,15 @@ public Void putPerson(HttpServletRequest req, HttpServletResponse resp) {
...
         try {
-            possibleConfig = appConfigFetcher.configurationFor(req);
-        } catch (ApiKeyNotFoundException | RevokedApiKeyException | InvalidIpForApiKeyException e) {
+            possibleApplication = appConfigFetcher.applicationFor(req);
+        } catch (InvalidApiKeyException e) {
             log.error("Problem with API key for request " + req.getRequestURL(), e);
             return error(req, resp, AtlasErrorSummary.forException(e));
         }
 
-        if (possibleConfig.isNothing()) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/PeopleWriteController.java]: We still need to check if the application is present and if not throw this error

File: src/main/java/org/atlasapi/query/v2/ScheduleController.java [hunk 8]
```diff
@@ -199,27 +198,20 @@ private boolean allNullOrEmpty(String... params) {
...
-            ApplicationConfiguration appConfig) {
+    private Set<Publisher> getPublishers(String publisher, Application application) {
         if (!Strings.isNullOrEmpty(publisher)) {
-            return Sets.intersection(publishersFrom(publisher), appConfig.getEnabledSources());
+            return Sets.intersection(publishersFrom(publisher), application.getConfiguration().getEnabledReadSources());
         }
-        if (appConfig.precedenceEnabled()) {
-            return ImmutableSet.of(appConfig.orderdPublishers().get(0));
+        if (application.getConfiguration().isPrecedenceEnabled()) {
+            return ImmutableSet.of(application.getConfiguration().getEnabledReadSources().asList().get(0));
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/ScheduleController.java]: `.asList()` will not give you the sources in the correct order. You need to sort them using the ordering first.

File: src/main/java/org/atlasapi/query/v2/TopicWriteController.java [hunk 9]
```diff
@@ -70,24 +68,14 @@ public TopicWriteController(ApplicationConfigurationFetcher appConfigFetcher, To
...
         try {
-            possibleConfig = appConfigFetcher.configurationFor(req);
-        } catch (ApiKeyNotFoundException | RevokedApiKeyException | InvalidIpForApiKeyException e) {
+            possibleApp = appConfigFetcher.applicationFor(req);
+        } catch (InvalidApiKeyException e) {
             return error(req, resp, AtlasErrorSummary.forException(e));
 
         }
 
-        if (possibleConfig.isNothing()) {
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/query/v2/TopicWriteController.java]: Ditto. We still have to check if the application is present and if not throw this error

File: src/main/java/org/atlasapi/remotesite/events/EventsUtility.java [hunk 10]
```diff
@@ -111,7 +111,7 @@ private Topic resolveOrCreateDbpediaTopic(String title, Topic.Type topicType, St
             return topic;
         }
         throw new IllegalStateException(String.format(
-                "Topic store failed to create Topic with namespace %s and value %s", 
+                "Topic store failed to createDefault Topic with namespace %s and value %s",
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/remotesite/events/EventsUtility.java]: Rename error

File: src/main/java/org/atlasapi/remotesite/metabroadcast/similar/DefaultSimilarContentProvider.java [hunk 11]
```diff
@@ -43,26 +42,24 @@
     private final TraitHashCalculator traitHashCalculator;
     private final AvailableItemsResolver availableItemsResolver;
     private final UpcomingItemsResolver upcomingItemsResolver;
-    private final ApplicationConfiguration appConfig = 
-            ApplicationConfiguration.defaultConfiguration()
-                                    .withSources(Maps.asMap(Publisher.all(), new Function<Publisher, SourceStatus>() {
+    private final Application application; // = DefaultApplication.createWithReads(Publisher.all().asList());
```
Comment (Dalamar42) [inline src/main/java/org/atlasapi/remotesite/metabroadcast/similar/DefaultSimilarContentProvider.java]: Commented out code

File: src/main/resources/config/environment.properties [hunk 12]
```diff
@@ -26,6 +26,9 @@ processing.mongo.writeConcern=
 #Are applicatiosn enabled
 applications.enabled=true
 
+#Applications service host
+applications.client.host=
```
Comment (Dalamar42) [inline src/main/resources/config/environment.properties]: This should be propagated to the K8 and puppet configs as well

File: src/test/java/org/atlasapi/equiv/query/MergeOnOutputQueryExecutorTest.java [hunk 13]
```diff
@@ -59,12 +71,19 @@ protected void setUp() throws Exception {
...
+                .thenReturn(configurationWithReads(Publisher.BBC));
+		when(mergePeopleApp.getConfiguration())
+                .thenReturn(configurationWithReads(Publisher.PA, Publisher.RADIO_TIMES));
 	}
 	
 	public void dontTestMergingBrands() throws Exception {
 		fail("refactor this test");
 //		MergeOnOutputQueryExecutor merger = new MergeOnOutputQueryExecutor(delegate(brand1, brand2, brand3));
-//		ContentQuery query = ContentQuery.MATCHES_EVERYTHING.copyWithApplicationConfiguration(new ApplicationConfiguration(null, ImmutableList.of(Publisher.YOUTUBE, Publisher.BBC)));
+//		ContentQuery query = ContentQuery.MATCHES_EVERYTHING.copyWithApplication(new ApplicationConfiguration(null, ImmutableList.of(Publisher.YOUTUBE, Publisher.BBC)));
```
Comment (Dalamar42) [inline src/test/java/org/atlasapi/equiv/query/MergeOnOutputQueryExecutorTest.java]: Commented out code. Kill with fire.

**Pull request closed without merging**